### ISSUE TITLE
Performance - Kick off username modal first

### ISSUE THREAD

**Issue by jimmcgaw** [cryptag/leapchat#66]
To make the app seem more responsive, kick off the user modal while the
longer-running crypto tasks run in the background getting the app ready.

**Comment by jimmcgaw** [comment]
This looks promising: http://threadsjs.readthedocs.io/en/latest/

**Comment by elimisteve** [comment]
@jimmcgaw Nice find. LeapChat will eventually also be an Electron app, so something that can do both in-browser stuff and desktop/Node stuff sounds good.

**Comment by elimisteve** [comment]
Startup/init feels slow to me only on Firefox on Android. Where does it feel slow to you?

**Comment by Spetastian** [comment]
Maybe we can utilize web workers for the long running crypto stuff?

**Comment by elimisteve** [comment]
@Spetastian Yes, we looked into web workers a bit -- I haven't used them in production -- and they seem like a good solution

**Comment by elimisteve** [comment]
@Spetastian Just looked into this further, and web workers sound even more awesome than I'd originally hoped/thought!

Great [browser support](https://caniuse.com/#feat=webworkers), and it sounds practical to spawn 4-16 workers when we need that many, which will likely be:

1. On app init (generating keypair from passphrase)
2. Right after app init (decrypting all the just-downloaded messages)
3. (Once we have multi-room support) Upon loading any room not recently viewed (similar to point 2, above)
4. (In Ensue/Pursuance) When loading a new pursuance's task hierarchy (and therefore a bunch of tasks -- potentially dozens)

**Comment by elimisteve** [comment]
The solution: https://github.com/dchest/nacl-stream-js/tree/master/demo

Turns out `nacl-stream` is already a miniLock dependency!

**Comment by elimisteve** [comment]
Partially solved this by adding gzip compression in b3ae8bce533b3e641b4fe18c332d1d8aee25dcc9
